library/common.h

```diff
@@ -57,4 +57,91 @@
  */
 #define MBEDTLS_ALLOW_PRIVATE_ACCESS
 
+/** Byte Reading Macros
+ *
+ * To tidy up code and save horizontal and vertical space, use byte
+ * reading macros to cast
+ */
+#define MBEDTLS_BYTE_0( x ) ( (uint8_t) ( ( x ) & 0xff ) )
+#define MBEDTLS_BYTE_1( x ) ( (uint8_t) ( ( ( x ) >> 8  ) & 0xff ) )
+#define MBEDTLS_BYTE_2( x ) ( (uint8_t) ( ( ( x ) >> 16 ) & 0xff ) )
+#define MBEDTLS_BYTE_3( x ) ( (uint8_t) ( ( ( x ) >> 24 ) & 0xff ) )
+
+/*
+ * 32-bit integer manipulation macros (big endian)
+ */
+#ifndef MBEDTLS_GET_UINT32_BE
+#define MBEDTLS_GET_UINT32_BE(n,b,i)                    \
+    do {                                                \
+        (n) = ( (uint32_t) (b)[(i)    ] << 24 )         \
+            | ( (uint32_t) (b)[(i) + 1] << 16 )         \
+            | ( (uint32_t) (b)[(i) + 2] <<  8 )         \
+            | ( (uint32_t) (b)[(i) + 3]       );        \
+    } while( 0 )
+#endif
+
+#ifndef MBEDTLS_PUT_UINT32_BE
+#define MBEDTLS_PUT_UINT32_BE(n,b,i)                    \
+    do {                                                \
+        (b)[(i)    ] = (unsigned char) ( (n) >> 24 );   \
+        (b)[(i) + 1] = (unsigned char) ( (n) >> 16 );   \
+        (b)[(i) + 2] = (unsigned char) ( (n) >>  8 );   \
+        (b)[(i) + 3] = (unsigned char) ( (n)       );   \
+    } while( 0 )
+#endif
+
+/*
+ * 32-bit integer manipulation macros (little endian)
+ */
+#ifndef MBEDTLS_GET_UINT32_LE
+#define MBEDTLS_GET_UINT32_LE(n,b,i)                    \
```

Comment:
I think in poly1305.c lines 293, 294 ... we see that the MBEDTLS_BYTES_TO_U32_LE version of this with the target integer not as a macro parameter is more flexible thus I'd be inclined to rather go with that version thus something like:
```C
#define MBEDTLS_GET_UINT32_LE( data, offset ) \
    ( (uint32_t) (data)[offset] \
      | ( (uint32_t) (data)[( offset ) + 1] << 8 )  \
      | ( (uint32_t) (data)[( offset ) + 2] << 16 )  \
      | ( (uint32_t) (data)[( offset ) + 3] << 24 )  \
    )
```
And the same for all MBEDTLS_GET_* macros.
The macro parameter names are clearer as well like this I'd say.